File: android-native-app/app/src/main/java/pl/edu/pb/androidnativeapp/data/model/Page.kt
package pl.edu.pb.androidnativeapp.data.model

import com.squareup.moshi.Json

data class Page(
    @field:Json(name = "info") val info: Info,
    @field:Json(name = "results") val results: List<CharacterEntity>,
) {
    data class Info(
        @field:Json(name = "count") val count: Int?,
        @field:Json(name = "pages") val pages: Int?,
        @field:Json(name = "next") val next: String?,
        @field:Json(name = "prev") val prev: String?,
    )
}

File: flutter-multiplatform-app/flutter_multiplatform_app/lib/data/model/page.dart
import 'package:flutter_multiplatform_app/data/model/character_entity.dart';
import 'package:flutter_multiplatform_app/data/model/info.dart';

final class Page {
  const Page({
    required this.info,
    required this.results,
  });

  final Info info;
  final List<CharacterEntity> results;
}
